src/path_calculator.rs

```diff
@@ -57,25 +57,40 @@ impl PathCalculator {
         graph: &FastGraph,
         start: NodeId,
         end: NodeId,
+    ) -> Option<ShortestPath> {
+        self.calc_path_multiple_endpoints(graph, vec![(start, 0)], end)
+    }
+
+    pub fn calc_path_multiple_endpoints(
+        &mut self,
+        graph: &FastGraph,
+        starts: Vec<(NodeId, Weight)>,
+        end: NodeId,
     ) -> Option<ShortestPath> {
         assert_eq!(
             graph.get_num_nodes(),
             self.num_nodes,
             "given graph has invalid node count"
         );
-        assert!(start < self.num_nodes, "invalid start node");
+        for (id, _) in &starts {
+            assert!(*id < self.num_nodes, "invalid start node");
+        }
         assert!(end < self.num_nodes, "invalid end node");
         self.heap_fwd.clear();
         self.heap_bwd.clear();
         self.valid_flags_fwd.invalidate_all();
         self.valid_flags_bwd.invalidate_all();
-        if start == end {
-            return Some(ShortestPath::singular(start));
+        for (id, _) in &starts {
+            if *id == end {
+                return Some(ShortestPath::singular(*id));
```

Comment:
I think here we should include the start weight. Say we route from node 3 with weight 6 to node 3 the result should be `nodes=[3], weight=6`, right?

Reply:
... I did